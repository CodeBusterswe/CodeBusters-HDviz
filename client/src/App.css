.frame {
  fill: none; /*riempimento*/
  stroke: grey; /*contorno del quadrato*/
}

.axis,
.frame {
  shape-rendering: crispEdges;
}

.axis line {
  stroke: #ddd;
}

.axis path {
  display: none;
}

.cell text {
  font-weight: bold;
  text-transform: capitalize;
  fill: black;
}

.frame {
  fill: none;
  stroke: #aaa;
}

.extent {
  fill: #000;
  fill-opacity: .125;
  stroke: #fff;
}
