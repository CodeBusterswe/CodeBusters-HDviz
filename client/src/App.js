import React, { useContext } from 'react'
import Store from './model/Model'
import ViewModel from './ViewModel'
import ViewController from './ViewController'

const App = () => {
  const store = useContext(Store);
  const viewModel = new ViewModel(store);
  return (
    <ViewController viewModel = {viewModel}/>
  )
}

export default App;
