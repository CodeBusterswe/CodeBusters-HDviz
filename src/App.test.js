import { render } from '@testing-library/react';
import ScatterPlotMatrix from "./components/SPM";
import App from './App';

const DimColore = 'color',
Data = [{sepallength: 5.1, sepalwidth: 3.5, petallength: 1.4, petalwidth: 0.2, class: "Iris-setosa"},
        {sepallength: 4.9, sepalwidth: 3, petallength: 1.4, petalwidth: 0.2, class: "Iris-setosa"}],
Dims = ["sepallength", "sepalwidth", "petallength", "petalwidth"];

//test App =====================================================================
test('app renderizzata correttamente', () => {
  render(<App />);
});

//test SPM =====================================================================

test('SPM renderizzato correttamente', () => {
  render(<ScatterPlotMatrix dimColor={DimColore} 
                            data={Data} 
                            dims={Dims.filter((d) => d!="-")} 
         />);
});

//snapshot test =================================================================
test("<SPM /> corrisponde allo snapshot", () => {
  //render ritorna un oggetto con diverse properties, una di queste Ã¨ il container
  const component = render(<ScatterPlotMatrix dimColor={DimColore} 
                                              data={Data} 
                                              dims={Dims.filter((d) => d!="-")} 
                            />);
  //container contiene il "DOM node" dell'elemento renderizzato (div)
  expect(component.container).toMatchSnapshot();
});

//test =================================================================