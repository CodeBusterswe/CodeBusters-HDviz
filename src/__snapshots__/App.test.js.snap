// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SPM /> corrisponde allo snapshot 1`] = `
<div>
  <div
    class="p-4"
  >
    <svg
      height="740"
      width="883"
    >
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(20,0)"
      >
        <path
          class="domain"
          d="M10.5,720V0.5H170.5V720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(10.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            4.90
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(50.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            4.95
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(90.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            5.00
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(130.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            5.05
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(170.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            5.10
          </text>
        </g>
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(200,0)"
      >
        <path
          class="domain"
          d="M10.5,720V0.5H170.5V720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(10.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            3.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(42.50000000000003,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            3.1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(74.50000000000006,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            3.2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(106.49999999999994,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            3.3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(138.49999999999997,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            3.4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(170.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            3.5
          </text>
        </g>
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(380,0)"
      >
        <path
          class="domain"
          d="M10.5,720V0.5H170.5V720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(90.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            1
          </text>
        </g>
      </g>
      <g
        class="x axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(560,0)"
      >
        <path
          class="domain"
          d="M10.5,720V0.5H170.5V720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(90.5,0)"
        >
          <line
            stroke="currentColor"
            y2="720"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="723"
          >
            0
          </text>
        </g>
      </g>
      <g
        class="y axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(20,0)"
      >
        <path
          class="domain"
          d="M720,170.5H0.5V10.5H720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,170.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            4.90
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,130.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            4.95
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,90.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            5.00
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,50.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            5.05
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,10.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            5.10
          </text>
        </g>
      </g>
      <g
        class="y axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(20,180)"
      >
        <path
          class="domain"
          d="M720,170.5H0.5V10.5H720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,170.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            3.0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,138.49999999999997)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            3.1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,106.49999999999994)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            3.2
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,74.50000000000006)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            3.3
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,42.50000000000003)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            3.4
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,10.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            3.5
          </text>
        </g>
      </g>
      <g
        class="y axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(20,360)"
      >
        <path
          class="domain"
          d="M720,170.5H0.5V10.5H720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,90.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            1
          </text>
        </g>
      </g>
      <g
        class="y axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(20,540)"
      >
        <path
          class="domain"
          d="M720,170.5H0.5V10.5H720"
          stroke="currentColor"
        />
        <g
          class="tick"
          opacity="1"
          transform="translate(0,90.5)"
        >
          <line
            stroke="currentColor"
            x2="720"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-3"
          >
            0
          </text>
        </g>
      </g>
      <g
        class="cell"
        transform="translate(20,0)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
        <text
          dy=".71em"
          x="20"
          y="20"
        >
          sepallength
        </text>
      </g>
      <g
        class="cell"
        transform="translate(20,180)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(20,360)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(20,540)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(200,0)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(200,180)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
        <text
          dy=".71em"
          x="20"
          y="20"
        >
          sepalwidth
        </text>
      </g>
      <g
        class="cell"
        transform="translate(200,360)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(200,540)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(380,0)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(380,180)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(380,360)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
        <text
          dy=".71em"
          x="20"
          y="20"
        >
          petallength
        </text>
      </g>
      <g
        class="cell"
        transform="translate(380,540)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(560,0)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(560,180)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(560,360)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
      </g>
      <g
        class="cell"
        transform="translate(560,540)"
      >
        <rect
          class="frame"
          height="160"
          width="160"
          x="10"
          y="10"
        />
        <circle />
        <circle />
        <text
          dy=".71em"
          x="20"
          y="20"
        >
          petalwidth
        </text>
      </g>
      <g
        class="legend"
        transform="translate(740,20)"
      >
        <rect
          height="18"
          style="fill: #1f77b4; stroke: #1f77b4;"
          width="18"
        />
        <text
          x="22"
          y="14"
        />
      </g>
    </svg>
  </div>
</div>
`;
